docs/how_to/extending_page_title.rst

```diff
@@ -134,37 +131,113 @@ Here's a simple version for our example::
+each language version of each ``Page``.
+
+..  note::
+
+    Please refer to the more detailed discussion above of the Page model extension example, and in particular to the
+    special **notes**.
+
+
+The model
+---------
+
+::
+
+    from django.db import models
+    from cms.extensions import TitleExtension
+    from cms.extensions.extension_pool import extension_pool
+
+
+    class RatingExtension(TitleExtension):
+        rating = models.IntegerField()
+
+
+    extension_pool.register(RatingExtension)
+
+
+The admin
+---------
+
+::
+
+    from django.contrib import admin
+    from cms.extensions import TitleExtensionAdmin
+    from .models import RatingExtension
+
+
+    class RatingExtensionAdmin(TitleExtensionAdmin):
+        pass
+
+
+    admin.site.register(RatingExtension, RatingExtensionAdmin)
+
+
+The toolbar item
+----------------
+
+In this example, we need to loop over the titles for the page, and populate the menu with those.
+
+::
+
+    from cms.toolbar_pool import toolbar_pool
+    from cms.extensions.toolbar import ExtensionToolbar
+    from django.utils.translation import ugettext_lazy as _
+    from .models import RatingExtension
+    from cms.utils import get_language_list  # needed to get the page's languages
+    @toolbar_pool.register
+    class RatingExtensionToolbar(ExtensionToolbar):
+        # defines the model for the current toolbar
+        model = RatingExtension
 
         def populate(self):
             # setup the extension toolbar with permissions and sanity checks
             current_page_menu = self._setup_extension_toolbar()
+
             # if it's all ok
             if current_page_menu and self.toolbar.edit_mode:
-                # create a sub menu
-                position = 0
-                sub_menu = self._get_sub_menu(current_page_menu, 'submenu_label', 'Submenu', position)
-                # retrieves the instances of the current title extension (if any) and the toolbar item URL
+                # create a sub menu labelled "Ratings" at position 1 in the menu
+                sub_menu = self._get_sub_menu(
+                    current_page_menu, 'submenu_label', 'Ratings', position=1
+                    )
+
+                # retrieves the instances of the current title extension (if any)
+                # and the toolbar item URL
                 urls = self.get_title_extension_admin()
-                # cycle through the title list
-                for title_extension, url in urls:
+
+                # we now also need to get the titleset (i.e. different language titles)
+                # for this page
+                page = self._get_page()
+                titleset = page.title_set.filter(language__in=get_language_list(page.site_id))
+
+                # create a 3-tuple of (title_extension, url, title)
+                nodes = [(title_extension, url, title.title) for (
+                    (title_extension, url), title) in zip(urls, titleset)
+                    ]
```

Comment:
This shouldn't be necessary! There ought to be a much nicer way to get the language versions' titles. I think it will need different behaviour from ``get_title_extension_admin()`` however - what do you think @czpython?